src/Domain.LinnApps/PurchaseOrders/PurchaseOrderDeliveryService.cs

```diff
@@ -77,11 +90,21 @@ public class PurchaseOrderDeliveryService : IPurchaseOrderDeliveryService
                 throw new UnauthorisedActionException("You are not authorised to acknowledge orders.");
             }
 
+            if (!this.purchaseLedgerMaster.GetRecord().OkToRaiseOrder.Equals("Y"))
+            {
+                throw new UnauthorisedActionException("Orders are currently restricted.");
+            }
+
             var entity = this.repository.FindById(key);
+            var miniOrder = this.miniOrderRepository.FindById(key.OrderNumber);
+            var miniOrderDelivery = this.miniOrderDeliveryRepository.FindBy(
+                x => x.OrderNumber == key.OrderNumber && x.DeliverySequence == key.DeliverySequence);
 
-            if (from.DateAdvised != to.DateAdvised)
+            if (from.AdvisedDate != to.AdvisedDate)
```

Comment:
I still feel funny about checking something other than the entity pulled from the db, in order to determine whether to update the entity from the db but we've already had a pr discussion about that 😄

Reply:
Yeah - will go over at show & tell and we can discuss